### PR TITLE
Feat(domains): Add Anycast DNS info and export option

### PR DESCRIPTION
**Pull request by rfougier**
### Requirements

* 2API `2i.2api.endpoints.sws/sws-domains` update has to be delivered before this feature

### Description of the Change
- add new column DNS Anycast in all domains table
- add new CSV export DNS Anycast option

### CODE REVIEW

File: client/app/resources/i18n/domains/Messages_fr_FR.xml [hunk 1]
```diff
@@ -22,7 +23,7 @@
    <translation id="domains_dashboard_tooltip_whois_multiples" qtlid="143121">Configurer le WHOIS des domaines sélectionnés</translation>
    <translation id="domains_dashboard_tooltip_dnssec_multiples" qtlid="184542">Configurer DNSSEC des domaines sélectionnés</translation>
    <translation id="domains_dashboard_tooltip_export_csv" qtlid="231682">Exporter les domaines en format CSV</translation>
-   <translation id="domains_dashboard_tooltip_export_csv_expl" qtlid="273046">L'export de vos domaines peut durer quelques minutes (en fonction du nombre de domaines que vous possédez). Durant l’opération, vous pouvez utiliser normalement le manager en restant dans la partie web. Le téléchargement va s’éffectuer automatiquement.</translation>
+   <translation id="domains_dashboard_tooltip_export_csv_expl" qtlid="273046">L'export de vos domaines peut durer quelques minutes (en fonction du nombre de domaines que vous possédez). Durant l'opération, vous pouvez utiliser normalement le manager en restant dans la partie web. Le téléchargement va s'éffectuer automatiquement.</translation>
```
Comment (antleblanc) [inline client/app/resources/i18n/domains/Messages_fr_FR.xml]: Why are you doing this type change?

File: client/app/domain/domain.service.js [hunk 2]
```diff
@@ -1314,6 +1314,22 @@ angular.module("services").service(
...
 
         /**
+         * Get DNS Anycast status
+         * @param {string} serviceName
+         */
+        getDnsAnycast (serviceName) {
+            return this.OvhHttp
+                .get(`/domain/zone/${serviceName}`, { rootPath: "apiv6" })
+                .then((response) => {
+                    if (response.hasDnsAnycast != null) {
```
Comment (FredericEspiau) [inline client/app/domain/domain.service.js]: Some people don't like `!= null`, you should use instead`_(response.hasDnsAnycast).isBoolean()`

File: client/app/domain/domain.service.js [hunk 3]
```diff
@@ -1314,6 +1314,22 @@ angular.module("services").service(
...
+                        return { status: (response.hasDnsAnycast && "enabled") || "disabled" };
```
Comment (FredericEspiau) [inline client/app/domain/domain.service.js]: Not a big fan of this kind of thing, I feel like it's less readable than `response.hasDnsAnycast ? "enabled" : "disabled"`

File: client/app/domain/domain.service.js [hunk 4]
```diff
@@ -1314,6 +1314,22 @@ angular.module("services").service(
...
+                    }
+                    return null;
+                })
+                .catch((err) => this.$q.reject(err));
```
Comment (FredericEspiau) [inline client/app/domain/domain.service.js]: Can be removed

File: client/app/domains/domains.html [hunk 5]
```diff
@@ -91,16 +92,32 @@ <h1 data-i18n-static="domains_dashboard_title"></h1>
...
+                                              data-ng-if="domain.whoisFields.length === 0"></span>
                                     </td>
                                     <td class="text-center">
-                                        <span data-ng-if="domain.dnssecSupported && domain.dnssecStatus !== null"
-                                              data-ng-bind="::i18n['domains_dashboard_table_dnssec_status_' + domain.dnssecStatus]"></span>
-                                        <span data-ng-if="!domain.dnssecSupported"
-                                              data-i18n-static="common_not_supported"></span>
+                                        <span class="label"
+                                              data-ng-bind="::i18n['domains_dashboard_table_dnssec_status_' + domain.dnssecStatus]"
+                                              data-ng-class="{'label-success': domain.dnssecStatus === 'ENABLED', 'label-danger': domain.dnssecStatus === 'DISABLED', 'label-info': domain.dnssecStatus !== 'ENABLED' && domain.dnssecStatus !== 'DISABLED'}"
```
Comment (FredericEspiau) [inline client/app/domains/domains.html]: It looks like `domain.dnssecStatus` could be a boolean and the text just above could be computed according to the truthness or falsness of the statement

File: client/app/domains/domains.html [hunk 6]
```diff
@@ -91,16 +92,32 @@ <h1 data-i18n-static="domains_dashboard_title"></h1>
...
+                                              data-ng-if="domain.dnssecSupported && domain.dnssecStatus !== null"></span>
```
Comment (FredericEspiau) [inline client/app/domains/domains.html]: I like `angular.isString(domain.dnssecStatus)` better but sometimes it doesn't work :(

File: client/app/domains/export-csv/domains-export-to-csv.controller.js [hunk 7]
```diff
@@ -37,6 +37,7 @@ angular.module("App").controller(
                 { label: "expiration_date", modelKey: "expiration", checked: true },
                 { label: "whois_fields", modelKey: "owo", checked: false, target: "owo", transform: (value, translations) => _.size(value) ? translations.enabled : translations.disabled },
                 { label: "dnssec", modelKey: "dnssec", checked: false, target: "dnssec", transform: (value, translations) => value && value.status ? translations[value.status] : "" },
+                { label: "dnsanycast", modelKey: "dnsanycast", checked: false, target: "dnsanycast", transform: (value, translations) => value && value.status ? translations[value.status] : "" },
```
Comment (FredericEspiau) [inline client/app/domains/export-csv/domains-export-to-csv.controller.js]: `_(value.status).isString() ? translations[value.status] : ""`

File: client/app/resources/i18n/domains/Messages_fr_FR.xml (hunk 1 above)
Comment (FredericEspiau) [inline client/app/resources/i18n/domains/Messages_fr_FR.xml]: `manager` => `espace client`

Comment (FredericEspiau) [inline client/app/resources/i18n/domains/Messages_fr_FR.xml]: `éffectuer` => `effectuer`

Comment (FredericEspiau) [reply client/app/resources/i18n/domains/Messages_fr_FR.xml]: `’` isn't a correct apostophe in French so he is right

File: client/app/resources/i18n/domains/Messages_fr_FR.xml [hunk 8]
```diff
@@ -55,10 +56,10 @@
...
    <translation id="domains_configuration_whois_step1_helpwizard_whatinfo" qtlid="130111">Quelles informations sont publiées sur le WHOIS ?</translation>
-   <translation id="domains_configuration_whois_step1_helpwizard_whatinfo_desc" qtlid="329512">Les informations publiées sont celles du propriétaire du nom de domaine. Elles dépendent de l’extension du domaine, mais on retrouve généralement :<ul><li>Son adresse</li><li>Ses numéros de téléphone et fax</li><li>Son e-mail</li></ul><br/>Le masquage/affichage peut être configuré indifféremment pour ces 3 types d’informations.</translation>
+   <translation id="domains_configuration_whois_step1_helpwizard_whatinfo_desc" qtlid="329512">Les informations publiées sont celles du propriétaire du nom de domaine. Elles dépendent de l'extension du domaine, mais on retrouve généralement :<ul><li>Son adresse</li><li>Ses numéros de téléphone et fax</li><li>Son e-mail</li></ul><br/>Le masquage/affichage peut être configuré indifféremment pour ces 3 types d'informations.</translation>
```
Comment (FredericEspiau) [inline client/app/resources/i18n/domains/Messages_fr_FR.xml]: Am I the only one who thinks this sentence is very badly written ?

**Pull request merged**